orchard/modules/_driver.py

```diff
@@ -5,7 +5,100 @@
+                # Check that config file path has the correct extension
+                self.config_file_path = str(sys.argv[2])
+                if (not self.config_file_path.split('.')[1] == 'yaml' and
+                        not self.config_file_path.split('.')[1] == 'yml'):
+                    print('The configuration file path has',
+                          'the wrong extension.')
+                    print('\tExpected: \t.yaml')
+                    print('\tFound: \t\t.',
+                          self.config_file_path.split('.')[1])
+                    sys.exit()
+
+                # Check that the configuration file exists
+                if (not os.path.isfile(self.config_file_path)):
+                    print('The configuration file: ' +
+                          self.config_file_path + ' does not exist.')
+                    sys.exit()
+            else:
+                self.generate_config_file()
+        else:
+            print('Orchard requires a Link File path' +
+                  ' and/or a Config File path')
+            sys.exit()
+        print("Nothing Went Wrong")
+
+    def generate_config_file(self):
+        with open(self.link_file_path) as fileHandle:
+            try:
+                dictionary = yaml.load(fileHandle, Loader=yaml.Loader)
+            except Exception as e:
+                print("The link file is not in the correct format.")
+                sys.exit()
+            dict_list = dictionary['modules']
+            for modules in dict_list:
+                try:
+                    del modules['dependencies']
+                    del modules['exlusive']
+                except KeyError:
+                    pass
+                if ('arguments' in modules):
+                    for arguments in modules['arguments']:
+                        arguments[arguments['name']] = None
+                        arguments.pop('name')
+                        try:
+                            del arguments['isBranch']
+                        except KeyError:
+                            pass
+                        try:
+                            del arguments['command']
+                        except KeyError:
+                            pass
+                if ('optionals' in modules):
+                    for arguments in modules['optionals']:
+                        arguments[arguments['name']] = None
+                        arguments.pop('name')
+                        try:
+                            del arguments['isBranch']
+                        except KeyError:
+                            pass
+                        try:
+                            del arguments['command']
+                        except KeyError:
+                            pass
+                        try:
+                            del arguments['isFlag']
+                        except KeyError:
+                            pass
+            yaml.SafeDumper.add_representer(
```

Comment:
I think the comment on this one got collapsed during your first commit as its line number changed so GitHub thought it was resolved. I'll copy it over from the first pass.

---

pep8 makes this hard to read, thoughts on a separate function?

```python
def _representer(dumper, value):
    return dumper.represent_scalar(u'tag:yaml.org,2002:null', '')

yaml.SafeDumper.add_representer(type(None), _representer)
```

not too worried either way, just pep8 sucks sometimes